src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategyTransformer.kt

```diff
@@ -950,23 +950,24 @@ internal class ManagedStrategyTransformer(
     }
 
     /**
-     * Removes switch points in CancellableContinuationImpl.cancel, so that they will not be reported
-     * when a continuation is cancelled by lincheck
+     * Removes ignored section for `CancellableContinuationImpl.onCancellation` call, so that
+     * events in the `onCancellation` method influence the execution and the trace.
+     *
+     * Note that this transformer does not handle the case of cancel handler throwing an exception.
      */
     private inner class CancellabilitySupportMethodTransformer(methodName: String, mv: GeneratorAdapter) : ManagedStrategyMethodVisitor(methodName, mv) {
-        private val isCancel = className == "kotlinx/coroutines/CancellableContinuationImpl" &&
-            (methodName == "cancel" || methodName == "cancelCompletedResult")
+        private val isOnCancellation = className == "kotlinx/coroutines/InvokeOnCancel" && methodName == "invoke"
```

Comment:
This is a private class -- we can't rely on it.

Reply:
We could leave and enter the ignored section back for the whole cancellation, I guess

Reply:
I leaved ignored section here so as to log the least possible odd method calls.
I can do the same for the `CancellableContinutionImplcallCancelHandler`, but `onCancellation` call stack will look worse.

Now it is
```
invoke(Exception@1): Unit@1 at InvokeOnCancel.invoke(CancellableContinuationImpl.kt:539)
  invoke(Exception@1) at CoroutineCancellationTraceReportingTest$cancelledOp$$inlined$suspendCancellableCoroutine$lambda$1.invoke(CoroutineCancellationTraceReportingTest.kt:30)
    setCorrect(false) at CoroutineCancellationTraceReportingTest$cancelledOp$$inlined$suspendCancellableCoroutine$lambda$1.invoke(CoroutineCancellationTraceReportingTest.kt:39)
      correct.WRITE(false) at CoroutineCancellationTraceReportingTest.setCorrect(CoroutineCancellationTraceReportingTest.kt:32)
```
but will be
```
invoke(Exception@1) at CancellableContinuationImpl.callCancelHandler(CancellableContinuationImpl.kt:230)
  invoke(Exception@1): Unit@1 at InvokeOnCancel.invoke(CancellableContinuationImpl.kt:539)
    invoke(Exception@1) at CoroutineCancellationTraceReportingTest$cancelledOp$$inlined$suspendCancellableCoroutine$lambda$1.invoke(CoroutineCancellationTraceReportingTest.kt:30)
      setCorrect(false) at CoroutineCancellationTraceReportingTest$cancelledOp$$inlined$suspendCancellableCoroutine$lambda$1.invoke(CoroutineCancellationTraceReportingTest.kt:39)
        correct.WRITE(false) at CoroutineCancellationTraceReportingTest.setCorrect(CoroutineCancellationTraceReportingTest.kt:32)
```

By the way, it looks bad in both variants

Reply:
Well, now it is
```
_state.compareAndSet(InvokeOnCancel@1,CancelledContinuation@1): true at CancellableContinuationImpl.cancel(CancellableContinuationImpl.kt:193)
invoke(Exception@1) at CancellableContinuationImpl.callCancelHandler(CancellableContinuationImpl.kt:230)
  invoke(Exception@1): Unit@1 at InvokeOnCancel.invoke(CancellableContinuationImpl.kt:539)
    invoke(Exception@1) at CoroutineCancellationTraceReportingTest$cancelledOp$$inlined$suspendCancellableCoroutine$lambda$1.invoke(CoroutineCancellationTraceReportingTest.kt:30)
      setCorrect(false) at CoroutineCancellationTraceReportingTest$cancelledOp$$inlined$suspendCancellableCoroutine$lambda$1.invoke(CoroutineCancellationTraceReportingTest.kt:39)
        correct.WRITE(false) at CoroutineCancellationTraceReportingTest.setCorrect(CoroutineCancellationTraceReportingTest.kt:32)

```

Reply:
Too much `invoke`-s  😄